### PR TITLE
build(weaver): eliminate need for package-local.json for besu/node

### PR DESCRIPTION
**Pull request by krisstern**
Related to #2601. 

VSCode does not recognize package-local.json files semantically as package.json manifests meaning that the convenience functionalities it offers for editing package.json files are turned off.

It has been made sure that the original package-local.json has been renamed. The corresponding build scripts updated to delete the logic that copies/moves around package.json files at build time. The package.json files that are getting used can always be looked at just based on what's stored in version control instead of having to guess what changes are being made to the files by the build at runtime.

**Pull Request Requirements**
- [x] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [x] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [x] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [x] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [x] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Review by petermetz — CHANGES_REQUESTED** [review]
@krisstern Can this be done in a way that the `package-local.json` file can be deleted completely instead of having to be put under a separate directory? The way you did it here technically satisfies the requirement of having the VSCode package.json integration working, but it still leaves us with the maintenance burden of having to test everything with 2 package.json files.

**Comment by krisstern** [comment]
> @krisstern Can this be done in a way that the `package-local.json` file can be deleted completely instead of having to be put under a separate directory? The way you did it here technically satisfies the requirement of having the VSCode package.json integration working, but it still leaves us with the maintenance burden of having to test everything with 2 package.json files.

Hi @petermetz, yes I think that it is possible to have one "package.json" file only. Let me try it over the next few days and update the PR shortly after.

**Comment by petermetz** [comment]
@krisstern Thank you!

**Comment by sandeepnRES** [comment]
Hi @krisstern Thanks for this PR.
The CI workflow that tests Besu SDK build is failing.
Check logs here: https://github.com/hyperledger/cacti/actions/runs/9116432968/job/25065050908?pr=3269

File: weaver/sdks/besu/node/package.json [hunk 1]
```diff
@@ -39,5 +39,8 @@
   "publishConfig": {
     "registry": "https://npm.pkg.github.com/hyperledger"
   },
-  "tag": "latest"
+  "tag": "latest",
+  "overrides": {
+    "got": "12.4.1"
```
Comment (sandeepnRES) [inline weaver/sdks/besu/node/package.json:44]: Any reason to add this `got` dependency?

Comment (krisstern) [reply weaver/sdks/besu/node/package.json:44]: HI @sandeepnRES, it's because the code would not compile otherwise.

Comment (petermetz) [reply weaver/sdks/besu/node/package.json:44]: @krisstern Sounds like a valid reason but still a little short on the details though. Is it the ESM cut-over that's causing it where later versions of `got` are ESM-only?
If yes, it warrants some more investigation to see if the original declaration could be changed instead for example.
The larger point  being: Just getting it to compile isn't always enough. We have to make sure that the solution does not make the code less maintainable through technical debt (which using the overrides definitely **is**)

Comment (krisstern) [reply weaver/sdks/besu/node/package.json:44]: Hi @petermetz Would that be acceptable if I update some of the versions for the packages in the `package.json`?

Comment (sandeepnRES) [reply weaver/sdks/besu/node/package.json:44]: @krisstern yes it perfectly fine to update the versions for packages as long as it builds and works as intended (for which CI tests are there, and one of them is failing right now)

Comment (sandeepnRES) [reply weaver/sdks/besu/node/package.json:44]: But on another point @krisstern I don't know what `got` package does but the code on main branch is stable and builds fine without the `got` package, 

So are you saying that it is required for this solution of removing `package-local.json`?

Comment (krisstern) [reply weaver/sdks/besu/node/package.json:44]: Hi @sandeepnRES It could also be due to my setup. May I ask which `Node` version you are using for the project?

Comment (sandeepnRES) [reply weaver/sdks/besu/node/package.json:44]: CI tests are using node 18

Comment (krisstern) [reply weaver/sdks/besu/node/package.json:44]: Yeah, I am using `Node 18` for the project too. So I am not sure why it didn't work for me before. Let me try again and see if I can do away with the override.

**Review by petermetz — CHANGES_REQUESTED** [review]

File: weaver/sdks/besu/node/package.json (hunk 1 above)
Comment (krisstern) [reply weaver/sdks/besu/node/package.json:44]: Yes, thanks @petermetz!

**Comment by petermetz** [comment]
@krisstern Are you still working on this? If not, that's totally fine, but then please let me know because I'm doing some housecleaning and this is one of the oldest pull requests we have open so if it's not happening I'll just close it down.

**Comment by krisstern** [comment]
Hi @petermetz, I may need more time working on this. But let me close this now and submit another PR soon to replace it.

**Comment by petermetz** [comment]
@krisstern Got it thank you. One very important thing: **Please do not open another PR. Re-open this one so that the context, review and commit history is intact**.

**Comment by krisstern** [comment]
Okay, noted!
